### ISSUE TITLE
Hyprland crashes when all monitors disconnected using hardware KVM

### ISSUE THREAD

**Issue by endeavour** [hyprwm/Hyprland#5309]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch HEAD at commit c5e28ebcfe00a510922779b2c568cfa52a317445  (props: bump ver 0.37.1).
Date: Sat Mar 16 14:51:49 2024
Tag: v0.37.1

flags: (if any)


System Information:
System name: Linux
Node name: office-desktop
Release: 6.7.9-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Fri, 08 Mar 2024 01:59:01 +0000


GPU information: 
0e:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP102 [GeForce GTX 1080 Ti] [10de:1b06] (rev a1) (prog-if 00 [VGA controller])


os-release: NAME="EndeavourOS"
PRETTY_NAME="EndeavourOS"
ID="endeavouros"
ID_LIKE="arch"
BUILD_ID="2022.04.08"
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://endeavouros.com"
DOCUMENTATION_URL="https://discovery.endeavouros.com"
SUPPORT_URL="https://forum.endeavouros.com"
BUG_REPORT_URL="https://forum.endeavouros.com/c/arch-based-related-questions/bug-reports"
PRIVACY_POLICY_URL="https://endeavouros.com/privacy-policy-2"
LOGO="endeavouros"


plugins:
```
</details>

### Bug or Regression?

Bug

### Description

I have two 4K monitors connected via a hardware displayport KVM. If I switch to my other system using the KVM, then hyprland crashes so I am unable to switch back.

Journalctl log and hyprland crash report are attached

### How to reproduce

I have found a workaround, which is to turn off one monitor before switching to my other system with the KVM. If I do this hyprland doesn't crash.

Therefore this bug only seems to happen if both monitors are disconnected at the same time. Perhaps it's a race condition?

### Crash reports, logs, images, videos

[crash.txt](https://github.com/hyprwm/Hyprland/files/14790129/crash.txt)
[log.txt](https://github.com/hyprwm/Hyprland/files/14790130/log.txt)

**Comment by vaxerski** [comment]
try on git

**Comment by endeavour** [comment]
Thanks, can confirmed this is fixed on v0.37.1-78-g187caf41 :) Keep up the good work, loving hyprland!
